xarray/tests/test_backends.py

```diff
@@ -992,6 +999,46 @@ def test_encoding_unlimited_dims(self):
             with pytest.warns(UserWarning):
                 ds.to_netcdf(tmp_file, engine='h5netcdf', unlimited_dims=['y'])
 
+class OpenMFDatasetTest(TestCase):
+    def validate_open_mfdataset_large_num_files(self, engine, nfiles=2000):
```

Comment:
Thanks @rabernat!  This context manager seems to do the trick for your future reference:

```python
@contextlib.contextmanager
def limit_open_num_files(nfiles=9):
    currentlim = resource.getrlimit(resource.RLIMIT_NOFILE)
    try:
        resource.setrlimit(resource.RLIMIT_NOFILE, 
                           (nfiles, resource.RLIM_INFINTY))
        yield
    finally:
        resource.setrlimit(resource.RLIMIT_NOFILE, currentlim)
        yield
```

usage 

```python
with limit_open_num_files():
     ...
```